src/main/java/org/zalando/nakadi/service/job/ExclusiveJobWrapper.java

```diff
@@ -0,0 +1,132 @@
+package org.zalando.nakadi.service.job;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Charsets;
+import org.apache.curator.utils.ZKPaths;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
+
+/**
+ * Class helps to run jobs that should be performed periodically only on single Nakadi instance
+ */
+public class ExclusiveJobWrapper {
+
+    public static final String NAKADI_JOBS_PATH = "/nakadi/jobs";
+
+    private final Logger log;
+    private final String zkPath;
+    private final ZooKeeperHolder zkHolder;
+    private final ObjectMapper objectMapper;
+    private final long jobPeriodMs;
+
+    public ExclusiveJobWrapper(final ZooKeeperHolder zkHolder,
+                               final ObjectMapper objectMapper,
+                               final String jobName,
+                               final long jobPeriodMs) {
+        this.zkHolder = zkHolder;
+        this.objectMapper = objectMapper;
+        this.jobPeriodMs = jobPeriodMs;
+        this.zkPath = ZKPaths.makePath(NAKADI_JOBS_PATH, jobName);
+        this.log = LoggerFactory.getLogger("nakadi-job." + jobName);
+    }
+
+    public void runJobLocked(final Runnable action) {
+        log.info("Trying to run job...");
+
+        if (!timeToRunAction()) {
```

Comment:
Could you please elaborate on the approach to run job at certain time as well as checking that it is time to run the action? I am thinking of something like, run jobs at certain time and see if there is a lock, then act or not.